src/pages/StocktakeEditPage.js

```diff
@@ -1,564 +1,224 @@
-    if (isModalOpen) return;
-    const quantity = parsePositiveInteger(newValue);
-    if (quantity === null) return;
-    stocktakeItem.setCountedTotalQuantity(database, quantity);
-
-    if (reasons.length > 0) this.assignReason(stocktakeItem);
-  };
-
-  /**
-   * Will reset all items contained in |this.itemsBelowMinimum| and
-   * |this.itemsOutdated|.
-   */
-  onResetItemsConfirm = () => {
-    const { database, stocktake, runWithLoadingIndicator } = this.props;
-    this.setState({ isResetModalOpen: false });
-    runWithLoadingIndicator(() => {
-      stocktake.resetStocktakeItems(database, this.itemsOutdated);
-      this.refreshData();
-    });
-  };
-
-  getModalTitle = () => {
-    const { modalKey } = this.state;
-    const { NAME_EDIT, COMMENT_EDIT, REASON_EDIT } = MODAL_KEYS;
+export const StocktakeEditPage = ({
+  runWithLoadingIndicator,
+  stocktake,
+  routeName,
+  dispatch: reduxDispatch,
+}) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(routeName, {
+    pageObject: stocktake,
+    backingData: stocktake.items,
+    data: stocktake.items.sorted('item.name').slice(),
+    keyExtractor: recordKeyExtractor,
+    dataState: new Map(),
+    currentFocusedRowKey: null,
+    searchTerm: '',
+    filterDataKeys: ['item.name'],
+    sortBy: 'itemName',
+    isAscending: true,
+    modalKey: '',
+    modalValue: null,
+  });
 
-    switch (modalKey) {
-      default:
-      case NAME_EDIT:
-        return modalStrings.edit_the_stocktake_name;
-      case COMMENT_EDIT:
-        return modalStrings.edit_the_stocktake_comment;
-      case REASON_EDIT:
-        return programStrings.select_a_reason;
+  const { STOCKTAKE_COMMENT_EDIT, EDIT_STOCKTAKE_BATCH, STOCKTAKE_OUTDATED_ITEM } = MODAL_KEYS;
+  const {
+    data,
+    dataState,
+    sortBy,
+    isAscending,
+    columns,
+    modalKey,
+    pageInfo,
+    pageObject,
+    currentStocktakeItem,
+    modalValue,
+    keyExtractor,
+  } = state;
+
+  const { isFinalised, comment, program } = pageObject;
+
+  useEffect(() => {
+    if (stocktake.itemsOutdated.length && !isFinalised) {
+      dispatch(openModal(STOCKTAKE_OUTDATED_ITEM));
     }
-  };
-
-  openModal = (key, stocktakeItem) =>
-    this.setState({ modalKey: key, isModalOpen: true, currentStocktakeItem: stocktakeItem });
+  }, []);
 
-  closeModal = () => this.setState({ isModalOpen: false });
-
-  updateDataFilters = (newSearchTerm, newSortBy, newIsAscending) => {
-    // (... != null) checks for null or undefined (implicitly type coerced to null).
-    if (newSearchTerm != null) this.dataFilters.searchTerm = newSearchTerm;
-    if (newSortBy != null) this.dataFilters.sortBy = newSortBy;
-    if (newIsAscending != null) this.dataFilters.isAscending = newIsAscending;
-  };
+  const renderPageInfo = useCallback(
+    () => <PageInfo columns={pageInfo(pageObject, dispatch)} isEditingDisabled={isFinalised} />,
```

Comment:
that columns <=> pageInfo naming thing again. just noting for a later PR possibly